docs/developers/tutorials/staking-contract.md

```diff
@@ -0,0 +1,415 @@
+
+After the building has completed, our folder should look like this:  
+![img](/developers/staking-contract-tutorial-img/folder_structure_2.png)
+
+A new folder, called `output` was created, which contains the compiled contract code. More on this is used later. For now, let's continue.
+
+# Your first lines of Rust
+
+Currently, we just have an empty contract. Not very useful, is it? So let's add some simple code for it. Since this is a staking contract, we'd expect to have a `stake` function, right?
+
+Let's add said function:
+```rust
+#[payable("EGLD")]
+#[endpoint]
+fn stake(&self) {}
+```
+
+Since we want this function to be callable by users, we have to annotate it with `#[endpoint]`. Also,since we want to be able to receive a payment, we mark it also as `#[payable("EGLD)]`. For now, we'll use EGLD as our staking token. Note that the contract does NOT need to be payable for it to receive payments on endpoint calls. The payable flag at contract level is only for receiving payments without endpoint invocation.
+
+Now, it's time to add an implementation for the function. We need to see how much a user paid, and save their staking information in storage. We end up with this code:
+```rust
+#[elrond_wasm::contract]
+pub trait StakingContract {
+    #[init]
+    fn init(&self) {}
+
+    #[payable("EGLD")]
+    #[endpoint]
+    fn stake(&self) {
+        let payment_amount = self.call_value().egld_value();
+        require!(payment_amount > 0, "Must pay more than 0");
+
+        let caller = self.blockchain().get_caller();
+        self.staking_position(caller.clone()).set(&payment_amount);
+        self.staked_addresses().insert(caller.clone());
+    }
+
+    #[view(getStakedAddresses)]
+    #[storage_mapper("stakedAddresses")]
+    fn staked_addresses(&self) -> UnorderedSetMapper<ManagedAddress>;
+
+    #[view(getStakingPosition)]
+    #[storage_mapper("stakingPosition")]
+    fn staking_position(&self, addr: ManagedAddress) -> SingleValueMapper<BigUint>;
+}
+```
+
+`require!` is a macro that is a shortcut for `if !condition { signal_error(msg) }`. Signalling an error will terminate the execution and revert any changes made to the internal state, including token transfers from and to the SC. In this case, there is no reason to continue if the user did not pay anything.
+
+We've also added #[view] annotation for the storage mappers, so we can later perform queries on those storage entries. You can read more about annotations here: https://docs.elrond.com/developers/developer-reference/elrond-wasm-annotations/
+
+Also, if you're confused about some of the functions used or the storage mappers, you can read more here:
+- https://docs.elrond.com/developers/developer-reference/elrond-wasm-api-functions/
+- https://docs.elrond.com/developers/developer-reference/storage-mappers/
+
+Now, I've intentionally written some bad code here. Can you spot the improvements we can make?
```

Comment:
I think it would be a lot useful to make a separate section names something like `useful practices` where you can enumerate some of these should do's in a RUST SC and probably arrange them in a certain manner with something like  `Don't do this` and `Do this`